Gems/ScriptCanvas/Code/CMakeLists.txt

```diff
@@ -89,6 +89,22 @@ ly_add_target(
         Gem::ScriptCanvasDebugger
 )
 
+ly_add_target(
+    NAME ScriptCanvasAPI HEADERONLY
+    NAMESPACE Gem
+    FILES_CMAKE
+        scriptcanvasgem_headers.cmake
+    COMPILE_DEFINITIONS
+        INTERFACE
+            SCRIPTCANVAS_ERRORS_ENABLED
+            ${SCRIPT_CANVAS_COMMON_DEFINES}
+    INCLUDE_DIRECTORIES
+        INTERFACE
+            .
+            Include
+            Include/ScriptCanvas
```

Comment:
I would expect this one to not be in the list, otherwise a client doesnt have to add the `ScriptCanvas` part of the include. Clients should always do `#include <ScriptCanvas/...`

Reply:
I also think `${SCRIPT_CANVAS_AUTOGEN_BUILD_DIR}` should be part of the interface, some of those files are being used by the multiplayer project

Reply:
Addressed here: https://github.com/o3de/o3de/pull/6012